arcgis-ios-sdk-samples/Layers/List KML contents/ListKMLContentsSceneViewController.swift

```diff
@@ -0,0 +1,168 @@
+// Copyright 2018 Esri.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import UIKit
+import ArcGIS
+
+class ListKMLContentsSceneViewController: UIViewController {
+    
+    @IBOutlet weak var sceneView: AGSSceneView!
+    
+    weak var kmlDataset: AGSKMLDataset?
```

Comment:
Neither of these need to be `weak` and they do not hold a strong reference to the view controller or anything with a strong reference to the view controller. In general, a view controller holds a strong reference to its data model.

Reply:
They're already retained by the parent view controller but I guess there's no harm in making them strong references.